lib/act_as_attachable.rb

```diff
@@ -0,0 +1,62 @@
+module ActAsAttachable
+  extend ActiveSupport::Concern
+
+  included do
+    ActiveRecord::Base.send(:extend, ModelClassMethods)
+    ActionController::Base.send(:extend, ControllerClassMethods)
+    ActionController::Base.send(:include, ControllerInstanceMethods)
+  end
+
+  module ModelClassMethods
+    # should be declared in model, then it will be able to have attachments
+    def act_as_attachable
+      has_many :attachments, as: :attachable
+
+      accepts_nested_attributes_for :attachments, allow_destroy: true,
+                                    reject_if: lambda { |params| params[:file_upload].blank? }
+    end
+  end
+
+  module ControllerInstanceMethods
+    # permit this in strong parameters
+    def attachments_params
+      { attachments_attributes: [:id, :file_upload, :file_upload_cache, :_destroy] }
+    end
+  end
+
+  module ControllerClassMethods
+    # should be declared in controller, to let it accept attachments
+    def accept_attachments
+      before_action :build_attachments, only: [:new, :edit]
+    end
+  end
+
+  protected
+
+  def build_attachments
+    @attachable = existing_instance || find_instance
+    @attachable.attachments.build if @attachable
+  end
+
+  def existing_instance
+    instance_variable_get('@' + attachable_item_name)
+  end
+
+  def find_instance
+    if params[:id]
+      attachable_item = klass_name.constantize.find(params[:id])
+    else
+      attachable_item = klass_name.constantize.new
+    end
+
+    instance_variable_set('@' + attachable_item_name, attachable_item)
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.